dataset_analysis/generate_dataframe.py

```diff
@@ -0,0 +1,462 @@
+
+    Input:
+        patient_data : dictionary containing information about the patient
+        dataset_path : path to the CanProCo dataset
+        timepoint : timepoint of the analysis (M0, M12)
+        output_folder : path to the output folder where the analysis will be saved
+    
+    Returns:
+        patient_data : dictionary containing information about the patient
+    """
+    #get the participant_id
+    participant_id = patient_data["participant_id"]
+
+    #now we find the lesion segmentation file and the spinal cord segmentation file
+    lesion_file_PISR = os.path.join(lesion_path, participant_id, "ses-" + timepoint, "anat", f"{participant_id}_ses-" + timepoint+ "_PSIR_lesion-manual.nii.gz")
+    if os.path.exists(lesion_file_PISR):
+        lesion_seg_file = lesion_file_PISR
+        sc_seg_file = os.path.join(lesion_path, participant_id, "ses-" + timepoint, "anat", f"{participant_id}_ses-" + timepoint + "_PSIR_seg-manual.nii.gz")
+    else:
+        # If PSIR doesn't exist, use STIR
+        lesion_seg_file = os.path.join(lesion_path, participant_id, "ses-" + timepoint, "anat", f"{participant_id}_ses-" + timepoint + "_STIR_lesion-manual.nii.gz")
+        sc_seg_file = os.path.join(lesion_path, participant_id, "ses-" + timepoint, "anat", f"{participant_id}_ses-" + timepoint + "_STIR_seg-manual.nii.gz")
+
+    #let's use sct to analyze the lesion
+    output_folder = os.path.join(output_folder, "tmp", participant_id)
+    os.system(f'sct_analyze_lesion -m {lesion_seg_file} -s {sc_seg_file} -ofolder {output_folder} -v 0')
+
+    #now we read the output file
+    output_file = str(lesion_seg_file).split('/')[-1].replace('.nii.gz', '_analysis.xls')
+    output_file = os.path.join(output_folder, output_file)
+    data = pd.read_excel(output_file)
+    
+    #we check if the data is empty
+    if data.empty:
+        #if it is empty we return the patient_data dictionary
+        return patient_data
+
+    number_of_lesions = max(data['label'])
+    total_lesion_volume = sum(data['volume [mm3]'])
+    biggest_lesion_vol = max(data['volume [mm3]'])
+    biggest_lesion_length = max(data['length [mm]'])
+    biggest_lesion_eq_diam = max(data['max_equivalent_diameter [mm]'])
+
+    #we add this information to the patient_data dictionary
+    patient_data["number_of_lesions"] = number_of_lesions
+    patient_data["total_lesion_volume"] = total_lesion_volume
+    patient_data["biggest_lesion_vol"] = biggest_lesion_vol
+    patient_data["biggest_lesion_length"] = biggest_lesion_length
+    patient_data["biggest_lesion_eq_diam"] = biggest_lesion_eq_diam
+
+    return patient_data
+
+
+def analyze_patient_tsv(participant_id, participants_tsv, timepoint):
+    """
+    This function gathers information from the participants.tsv file.
+    It gathers information on each participant, their pathology (and the material used for image acquisition).
```

Comment:
```suggestion
    It gathers information on each participant, their pathology (and the acquisition parameters used for image acquisition).
```

"material" --> "acquisition parameters"?